xla/stream_executor/device_description.h

```diff
@@ -159,33 +159,52 @@ class RocmComputeCapability {
     return absl::StrJoin(kSupportedGfxVersions, ", ");
   }
 
-  bool has_nhwc_layout_support() const {
-    static constexpr absl::string_view kList[] = {"gfx908", "gfx90a"};
+
+  bool gfx9_mi100_or_later() const {
+    static constexpr absl::string_view kList[] = {"gfx908", "gfx90a",
+                                                  "gfx940", "gfx941", "gfx942"};
     return absl::c_count(kList, gfx_version()) != 0;
   }
 
-  bool has_bf16_dtype_support() const {
-    static constexpr absl::string_view kList[] = {"gfx908", "gfx90a"};
+  bool gfx9_mi200_or_later() const {
+    static constexpr absl::string_view kList[] = {"gfx90a",
+                                                  "gfx940", "gfx941", "gfx942"};
     return absl::c_count(kList, gfx_version()) != 0;
   }
 
+  bool navi21() const { return gfx_version() == "gfx1030"; }
+
+  bool navi31() const { return gfx_version() == "gfx1100"; }
+
+  bool has_nhwc_layout_support() const {
+    return gfx9_mi100_or_later();
+  }
+
+  bool has_bf16_dtype_support() const {
+    return gfx9_mi100_or_later();
+  }
+
   bool has_fast_fp16_support() const {
-    static constexpr absl::string_view kList[] = {"gfx906", "gfx908", "gfx90a",
-                                                  "gfx1030"};
-    return absl::c_count(kList, gfx_version()) != 0;
+    return gfx9_mi100_or_later() || navi21() || navi31();
   }
 
   bool has_mfma_instr_support() const {
-    static constexpr absl::string_view kList[] = {"gfx908", "gfx90a"};
-    return absl::c_count(kList, gfx_version()) != 0;
+    return gfx9_mi100_or_later();
   }
 
   bool has_fp16_atomics_support() const {
     // TODO(rocm): Check. This should be the same as has_fast_fp16_support().
-    static constexpr absl::string_view kList[] = {"gfx90a"};
-    return absl::c_count(kList, gfx_version()) != 0;
+    return gfx9_mi200_or_later();
   }
 
+  bool fence_before_barrier() const {
+    return gfx_version() != "gfx900" && gfx_version() != "gfx906";
+  }
+
+  bool has_hipblaslt() const {
+    return gfx9_mi200_or_later();
+  };
```

Comment:
The linter is complaining about the ";" character in this line.
Also it is complaining that there is no newline after the last line in this file.